functions/imagemagick/index.js

```diff
@@ -30,115 +30,94 @@ const {BLURRED_BUCKET_NAME} = process.env;
+      detections.violence === 'VERY_LIKELY'
+    ) {
+      console.log(`The image ${file.name} has been detected as inappropriate.`);
+      return blurImage(file, BLURRED_BUCKET_NAME);
+    } else {
+      console.log(`The image ${file.name} has been detected as OK.`);
+    }
+  } catch (err) {
+    console.error(`Failed to analyze ${file.name}.`, err);
+    return Promise.reject(err);
+  }
 };
 // [END functions_imagemagick_analyze]
 
 // [START functions_imagemagick_blur]
 // Blurs the given file using ImageMagick, and uploads it to another bucket.
-function blurImage(file, blurredBucketName) {
+const blurImage = async (file, blurredBucketName) => {
   const tempLocalPath = `/tmp/${path.parse(file.name).base}`;
 
   // Download file from bucket.
-  return file
-    .download({destination: tempLocalPath})
-    .catch(err => {
-      console.error('Failed to download file.', err);
-      return Promise.reject(err);
-    })
-    .then(() => {
-      console.log(
-        `Image ${file.name} has been downloaded to ${tempLocalPath}.`
-      );
-
-      // Blur the image using ImageMagick.
-      return new Promise((resolve, reject) => {
-        gm(tempLocalPath)
-          .blur(0, 16)
-          .write(tempLocalPath, (err, stdout) => {
-            if (err) {
-              console.error('Failed to blur image.', err);
-              reject(err);
-            } else {
-              resolve(stdout);
-            }
-          });
-      });
-    })
-    .then(() => {
-      console.log(`Image ${file.name} has been blurred.`);
-
-      // Upload result to a different bucket, to avoid re-triggering this function.
-      // You can also re-upload it to the same bucket + tell your Cloud Function to
-      // ignore files marked as blurred (e.g. those with a "blurred" prefix)
-      const blurredBucket = storage.bucket(blurredBucketName);
-
-      // Upload the Blurred image back into the bucket.
-      return blurredBucket
-        .upload(tempLocalPath, {destination: file.name})
-        .catch(err => {
-          console.error('Failed to upload blurred image.', err);
-          return Promise.reject(err);
-        });
-    })
-    .then(() => {
-      console.log(
-        `Blurred image has been uploaded to: gs://${blurredBucketName}/${file.name}`
-      );
-
-      // Delete the temporary file.
-      return new Promise((resolve, reject) => {
-        fs.unlink(tempLocalPath, err => {
+  try {
+    await file.download({destination: tempLocalPath});
+
+    console.log(`Image ${file.name} has been downloaded to ${tempLocalPath}.`);
+  } catch (err) {
+    console.error('Failed to download file.', err);
+    return Promise.reject(err);
+  }
+
+  try {
+    // Blur the image using ImageMagick.
+    await new Promise((resolve, reject) => {
+      gm(tempLocalPath)
+        .blur(0, 16)
+        .write(tempLocalPath, (err, stdout) => {
           if (err) {
-            reject(err);
+            console.error('Failed to blur image.', err);
```

Comment:
This log statement is inside the promise resolution. Save this for the catch statement.

```suggestion
```